### PR TITLE
Add Kuryr-Kubernetes options and network type

### PR DESCRIPTION
**Pull request by dulek**
In order to support Kuryr in cluster-network-operator we need to add Kuryr stuff to operator API.

Please note that we still look if we'll be able to drop clusterId and find it elsewhere, so this is now WiP.

### CODE REVIEW

**Comment by dulek** [comment]
Okay, this is now ready for reviews, we found a way to get rid of ClusterID from the structure.

**Comment by dulek** [comment]
@openshift/api-reviewers - can you please take a look? We're waiting for this to proceed with work on cluster-network-operator.

File: operator/v1/types_network.go [hunk 1]
```diff
@@ -133,6 +137,20 @@ type OpenShiftSDNConfig struct {
...
 }
 
+// KuryrConfig configures the Kuryr-Kubernetes SDN
+type KuryrConfig struct {
+	//TODO(dulek): Remove ports from config once we're 100% sure that there
+	// won't be any conflicts (Kuryr runs with host networking).
+
+	// The port kuryr-daemon will listen for readiness and liveness requests.
+	// +optional
+	DaemonProbesPort *uint16 `json:"daemonProbesPort"`
```
Comment (wking) [inline operator/v1/types_network.go]: Optional properties should be `omitempty`.

Comment (dulek) [reply operator/v1/types_network.go]: Done, thanks!

**Comment by dulek** [comment]
@openshift/api-reviewers: I addressed @wking comment and I think this is ready.

File: operator/v1/types_network.go [hunk 2]
```diff
@@ -137,6 +141,17 @@ type OpenShiftSDNConfig struct {
 	UseExternalOpenvswitch *bool `json:"useExternalOpenvswitch,omitempty"`
 }
 
+// KuryrConfig configures the Kuryr-Kubernetes SDN
+type KuryrConfig struct {
+	// The port kuryr-daemon will listen for readiness and liveness requests.
+	// +optional
+	DaemonProbesPort *uint16 `json:"daemonProbesPort,omitempty"`
```
Comment (soltysh) [inline operator/v1/types_network.go]: Let's try to keep it consistent with other ports, such as:
- https://github.com/openshift/api/blob/master/operator/v1/types_network.go#L127
- https://github.com/openshift/api/blob/master/openshiftcontrolplane/v1/types.go#L223
- https://github.com/openshift/api/blob/master/operator/v1/types_network.go#L127
similarly in k8s API ports are usually 32-bit ints. I'm fine with this being uint,  but please make it 32-bit. Same applies to the other one.

Comment (dulek) [reply operator/v1/types_network.go]: Done, thanks!

**Pull request merged**